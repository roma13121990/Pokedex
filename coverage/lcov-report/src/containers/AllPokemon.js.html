<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/containers/AllPokemon.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/containers</a> AllPokemon.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/34</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/18</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a></td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React,<span class="cstat-no" title="statement not covered" > {Component} from 'react'<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ></span></span></span></span></span>
import {connect} f<span class="cstat-no" title="statement not covered" >rom 'react-redux'</span>
import WebApi <span class="cstat-no" title="statement not covered" >from "../utils/WebApi";<span class="cstat-no" title="statement not covered" ></span></span>
import PokemonListItem <span class="cstat-no" title="statement not covered" >from "../components/PokemonListItem";<span class="cstat-no" title="statement not covered" ></span></span>
import {Link} from "re<span class="cstat-no" title="statement not covered" >act-router-dom";</span>
import Loader from <span class="cstat-no" title="statement not covered" >'react-loader';<span class="cstat-no" title="statement not covered" ></span></span>
import {ToastContaine<span class="cstat-no" title="statement not covered" >r} from "react-toastify";<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ></span></span></span>
&nbsp;
export class AllPokemon extends Component {
<span class="fstat-no" title="function not covered" >    co</span>nstructor(props) {
<span class="cstat-no" title="statement not covered" >        super(props);<span class="cstat-no" title="statement not covered" ></span></span>
        this.state = {
            species: [],
            loading: false,
            allSpecies: [],
            myPokemon: []
        };
    }
&nbsp;
    componentDidMount() {
        WebApi.getListOfPokemon().then(res =&gt; res.json())
            .then(response =&gt; {
                this.setState({
                    species: response.results,
                    loading: true,
                    allSpecies: response.results
                });
            });
    }
&nbsp;
    filterList = (eve<span class="fstat-no" title="function not covered" >nt</span>) =&gt; {
        if (<span class="cstat-no" title="statement not covered" >event.target.value === "") this<span class="cstat-no" title="statement not covered" >.setState({species: this.state.allSpecies})</span></span>
        else {
            let updatedList = this<span class="cstat-no" title="statement not covered" >.state.species;</span>
            upda<span class="cstat-no" title="statement not covered" >tedList = updatedList.filter(func<span class="fstat-no" title="function not covered" >ti</span>on (item) {</span>
&nbsp;
                retu<span class="cstat-no" title="statement not covered" >rn item.name.toLowerCase().startsWith(</span>
                    event.target.value.toLowerCase());
            });
            this<span class="cstat-no" title="statement not covered" >.setState({species: updatedList});</span>
        }
    };<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span></span></span>
&nbsp;
<span class="fstat-no" title="function not covered" >    re</span>nder() {
<span class="cstat-no" title="statement not covered" >        const {load</span>ing, species} = <span class="cstat-no" title="statement not covered" >this.state;<span class="cstat-no" title="statement not covered" ></span></span>
&nbsp;
<span class="cstat-no" title="statement not covered" >        return (</span>
            &lt;Loader loaded={loading}&gt;
            &lt;div className="pokemon--species--search"&gt;
            &lt;p&gt;&lt;input type="text" className="form-control form-control-lg" placeholder="Search"
                      onChange={this.filterList}/&gt;
            &lt;/p&gt;
                &lt;ToastContainer /&gt;
            &lt;div className="pokemon--species--list"&gt;{species.map((pokemon, index) =&gt;<span class="fstat-no" title="function not covered" ></span>
                &lt;Link to<span class="cstat-no" title="statement not covered" >={`/PokemonDetails/${pokemon.name}`} key={index}&gt;</span>
                    {&lt;PokemonListItem key={pokemon.name} id={index + 1} pokemon={pokemon} showBtn="add"/&gt;}
                &lt;/Link&gt;
            )}&lt;/div&gt;
            &lt;/div&gt;&lt;/Loader&gt;
        )
    }<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ></span></span>
}
&nbsp;
export default connect()(AllPokemon)</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun May 06 2018 20:20:40 GMT+0530 (IST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
