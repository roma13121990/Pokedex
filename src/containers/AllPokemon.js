import React, {Component} from 'react'
import {connect} from 'react-redux'
import WebApi from "../utils/WebApi";
import PokemonListItem from "../components/PokemonListItem";
import {Link} from "react-router-dom";
import Loader from 'react-loader';
import {ToastContainer} from "react-toastify";

export class AllPokemon extends Component {
    constructor(props) {
        super(props);
        this.state = {
            species: [],
            loading: false,
            allSpecies: [],
            myPokemon: []
        };
    }

    componentDidMount() {
        //fetch a list of all pokemn
        WebApi.getListOfPokemon().then(res => res.json())
            .then(response => {
                this.setState({
                    species: response.results,
                    loading: true,
                    allSpecies: response.results
                });
            });
    }
    //Search feature which finds name in the state and update the state based on search patter
    filterList = (event) => {
        if (event.target.value === "") this.setState({species: this.state.allSpecies})
        else {
            let updatedList = this.state.species;
            updatedList = updatedList.filter(function (item) {

                return item.name.toLowerCase().startsWith(
                    event.target.value.toLowerCase());
            });
            this.setState({species: updatedList});
        }
    };

    render() {
        const {loading, species} = this.state;

        return (
            <Loader loaded={loading}>
            <div className="pokemon--species--search">
            <p><input type="text" className="form-control form-control-lg" placeholder="Search"
                      onChange={this.filterList}/>
            </p>
                <ToastContainer />
            <div className="pokemon--species--list">{species.map((pokemon, index) =>
                <Link to={`/PokemonDetails/${pokemon.name}`} key={index}>
                    {<PokemonListItem key={pokemon.name} id={index + 1} pokemon={pokemon} showBtn="add"/>}
                </Link>
            )}</div>
            </div></Loader>
        )
    }
}

export default connect()(AllPokemon)