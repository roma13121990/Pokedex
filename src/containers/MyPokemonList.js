import React from 'react'
import { connect } from 'react-redux'
import PokemonItem from "../components/PokemonListItem";
import {Link} from "react-router-dom";
const styles={"textAlign":"left","padding":"0 17px 5px 17px","color":"#827474"}

export const MyPokemonList = (props) => {
    console.log(props)
    return(
    <div className="list">
        <h4 style={styles}>My Pokemon</h4>
        <ul>
            {props.item.map((k,i) =>
            <Link to={`/PokemonDetails/${k.pokemon.name}`} key={i}>
                <PokemonItem
                    {...k} showBtn="remove"
                /></Link>
            )}
        </ul>
    </div>
)}

const mapStateToProps = state => ({
  item: state.item
})

export default connect(
  mapStateToProps
)(MyPokemonList)


