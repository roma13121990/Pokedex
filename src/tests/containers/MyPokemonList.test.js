import React from 'react';
import {shallow} from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import * as Enzyme from "enzyme";
import {MyPokemonList} from "../../containers/MyPokemonList";
import { MemoryRouter} from "react-router-dom";
import configureMockStore from "redux-mock-store";
import {Provider} from "react-redux";

const mockStore = configureMockStore();
const store = mockStore({});

Enzyme.configure({adapter: new Adapter()});
describe('<MyPokemonList />', () => {

    const props = {'item' : [
            {'pokemon':{'name':"Pikachoo"}},
            {'pokemon':{'name':"Abc"}},
            {'pokemon':{'name':"xyz"}}
        ]}

    it('should save snapshot for data render of fetch data from the store', () => {
        const wrapper = shallow(<Provider store={store}><MemoryRouter><MyPokemonList {...props} /></MemoryRouter></Provider>)

        expect(wrapper).toMatchSnapshot();

    })
})