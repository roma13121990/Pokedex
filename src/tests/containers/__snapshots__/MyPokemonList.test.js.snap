// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyPokemonList /> should save snapshot for data render of fetch data from the store 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
>
    <MemoryRouter>
        <Unknown
            item={
                Array [
                    Object {
                      "pokemon": Object {
                        "name": "Pikachoo",
                      },
                    },
                    Object {
                      "pokemon": Object {
                        "name": "Abc",
                      },
                    },
                    Object {
                      "pokemon": Object {
                        "name": "xyz",
                      },
                    },
                  ]
            }
        />
    </MemoryRouter>
</Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Unknown
        item={
                Array [
                        Object {
                          "pokemon": Object {
                            "name": "Pikachoo",
                          },
                        },
                        Object {
                          "pokemon": Object {
                            "name": "Abc",
                          },
                        },
                        Object {
                          "pokemon": Object {
                            "name": "xyz",
                          },
                        },
                      ]
        }
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "item": Array [
          Object {
            "pokemon": Object {
              "name": "Pikachoo",
            },
          },
          Object {
            "pokemon": Object {
              "name": "Abc",
            },
          },
          Object {
            "pokemon": Object {
              "name": "xyz",
            },
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Unknown
          item={
                    Array [
                              Object {
                                "pokemon": Object {
                                  "name": "Pikachoo",
                                },
                              },
                              Object {
                                "pokemon": Object {
                                  "name": "Abc",
                                },
                              },
                              Object {
                                "pokemon": Object {
                                  "name": "xyz",
                                },
                              },
                            ]
          }
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "item": Array [
            Object {
              "pokemon": Object {
                "name": "Pikachoo",
              },
            },
            Object {
              "pokemon": Object {
                "name": "Abc",
              },
            },
            Object {
              "pokemon": Object {
                "name": "xyz",
              },
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
